### ISSUE TITLE
LHS Hit Navigator redesign

### ISSUE THREAD

**Issue by mwinokan** [m2ms/fragalysis-frontend#1322]
Following the v2 release and lots of new tags it makes sense to revisit how to effectively display the new data.

**Comment by mwinokan** [comment]
@matej-vavrek could you start looking into this one you have finished with the outstanding purple release work?

**Comment by mwinokan** [comment]
Here's a suggestion for how to adjust the observations table to match:

![Screenshot 2024-02-05 at 09 59 56](https://github.com/m2ms/fragalysis-frontend/assets/36866506/6697d3ad-c4f1-4a45-a91f-ed2858ccc763)

Outstanding shortcomings/missing spec of the current proposal:

### 1. 8x1 vs 4x2 grid of [A][L][P][C] etc ... buttons

* In the current draft the hit navigator is the only place where the 4x2 layout is used, conceivably the RHS can be adjusted but it clashes with the narrow rows in the observations table.
* @phraenquex prefers the horizontal button layout but it underutilises vertical space.
* @matej-vavrek do you see a different solution for the hit navigator where we can keep the 8x1 layout?

### 2. Defining the colour cyclers/ranges

* @phraenquex How many colours do we need to support for canonical sites? If it's less than 10 we have many options. To support up to 20 we could use something like ColorBrewer2.0 qualitative tab10 and tab20 ([see here](https://proplot.readthedocs.io/en/latest/cycles.html#Included-color-cycles)), starting with the 10 bright colours and then using the lighter counterparts for the next 10:

<img width="437" alt="Screenshot 2024-02-05 at 09 51 45" src="https://github.com/m2ms/fragalysis-frontend/assets/36866506/d5819e53-5a7b-4225-b814-8721d0b899bc">

* The above colour cycler could be used for curator added tags?

* Then for the ConformerSites, CrystalForms, and Assemblies we can use a continuous range of colours, e.g.:

  - ConformerSites A range of greens
  - CrustalForms A grayscale range
  - Assemblies A range of oranges/browns

**Comment by boriskovar-m2ms** [comment]
@mwinokan currently the observations are grouped by compound and canonical site and it seems from your proposal that they should be grouped by compound, canonical site AND conformer site. Is this correct? Because currently this is possible (highlighted row of tags are conformer site)
![image](https://github.com/m2ms/fragalysis-frontend/assets/16878338/8f9582cc-3f20-40d0-a3f3-c4f6c3bf6b93)

**Comment by boriskovar-m2ms** [comment]
Aren't going be the canonical and conformer site tags exactly the same as they are on respective compound that makes it a redundant information?

**Comment by boriskovar-m2ms** [comment]
It seems that this proposal will make the LHS list a lot slicker but also slimmer which will affect Tag details window which is right above hit navigator so maybe we will need to redesign it too so they can continue to coexist peacefully in the same column.

**Comment by mwinokan** [comment]
@boriskovar-m2ms I just spoke to @phraenquex and we should not group by ConformerSite, but highlight the conformersite of the displayed observation. I will think of a way to indicate the diversity of conformers displayed across the observations

**Comment by mwinokan** [comment]
I will clarify the spec/mockup when I get a moment

**Comment by mwinokan** [comment]
@phraenquex @boriskovar-m2ms @matej-vavrek what do you think of the following to show diversity amongst observed ConformerSites in the hit navigator:

![Screenshot 2024-02-07 at 14 12 32](https://github.com/m2ms/fragalysis-frontend/assets/36866506/87247cd8-3a90-49c4-b3db-69bf04d8f88e)

If this is too busy I would suggest just showing the ConformerSite for the displayed/first observation

**Comment by boriskovar-m2ms** [comment]
@mwinokan I think it looks good

**Comment by phraenquex** [comment]
@mwinokan agree it looks great.  Run it past Daren and/or Ed too.  

Probably the conformer stacked column should be over to the right, just left of the orange observations block?

**Comment by mwinokan** [comment]
@matej-vavrek thanks for highlighting the difficulties with the 2x4 vertical arrangement of the NGL viewer buttons. Please could you explore the following layout:

![Screenshot 2024-03-12 at 11 53 37](https://github.com/m2ms/fragalysis-frontend/assets/36866506/05d9d364-d0ad-4183-8abd-18b7f6818e16)

- Do you think we would be very limited with the text sizes for the observations and compound? 
- And could we have a dynamic view for the tags so that they are shortened to the first X characters if the view is too narrow to accommodate them?

**Comment by mwinokan** [comment]
@matej-vavrek for #1261 could you please add radio buttons to the observations so that the 'main' one can be specified 

![Screenshot 2024-03-13 at 19 45 51](https://github.com/m2ms/fragalysis-frontend/assets/36866506/e8e34832-9469-4356-b251-64e6858a6348)

**Comment by boriskovar-m2ms** [comment]
@mwinokan for your last suggestion we need also support from backend if this should be target wise.

**Comment by mwinokan** [comment]
@matej-vavrek the radio buttons won't be needed until we decide what to do with #1261

**Comment by mwinokan** [comment]
Let's discuss at today's meeting, but here is a full mockup including all the various bits of spec discussed in this ticket and #1261 #1392

![Screenshot 2024-03-19 at 10 58 38](https://github.com/m2ms/fragalysis-frontend/assets/36866506/045a7954-0133-4ee8-a6ce-45e1765e3bff)

**Comment by boriskovar-m2ms** [comment]
> Let's discuss at today's meeting, but here is a full mockup including all the various bits of spec discussed in this ticket and #1261 #1392
> 
> ![Screenshot 2024-03-19 at 10 58 38](https://private-user-images.githubusercontent.com/36866506/314026073-045a7954-0133-4ee8-a6ce-45e1765e3bff.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxMjM4ODcsIm5iZiI6MTcxMjEyMzU4NywicGF0aCI6Ii8zNjg2NjUwNi8zMTQwMjYwNzMtMDQ1YTc5NTQtMDEzMy00ZWU4LWE2Y2UtNDVlMTc2NWUzYmZmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDA1NTMwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRjNzZiY2Y3NzFhZGUxODg2MjIzYjVkYzY3YzhmZGMwOWFjZjRhYWY1YTAyNTZhNzc1NDZhYzU0MWI2OWU5MzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.LCO3hNc7FSXDQERBM2-1Szx-_nGQipkRSlmQzuqc448)

@mwinokan  Here in the picture in the select ```x0123f``` and ```x0123k``` are already existing poses I guess. What if there is a lot of them in the select? How do we "delete" given pose? When we assign all of it's observations to different poses then the pose should be deleted? The pose name or ```display_name``` can be only name of one of it's observations or it can be any free form text? If former what if I add observations A to pose B and change it's name to A and then remove the observation from the pose?

**Comment by mwinokan** [comment]
@boriskovar-m2ms

> Here in the picture in the select x0123f and x0123k are already existing poses I guess.

Yes that's correct. There should be the option to add an observation to any existing other poses that share the same compound and canonical site.

> What if there is a lot of them in the select?

All selected observations should be added to the desired new or existing pose/group

> How do we "delete" given pose? When we assign all of its observations to different poses then the pose should be deleted?

Yes, there is no point having a pose with no observations.

> The pose name or display_name can be only name of one of its observations or it can be any free form text? If former what if I add observations A to pose B and change its name to A and then remove the observation from the pose?

The pose name should indeed just be the name of an existing observation (for now), and in the case where the observation that provides the name is removed from a pose, the user should ideally rename the pose to a different observation. I think it's ok to leave that up to the curators for now, but @phraenquex may have thoughts.

**Comment by boriskovar-m2ms** [comment]
> > What if there is a lot of them in the select?
> 
> All selected observations should be added to the desired new or existing pose/group

By the select I meant the select-box or drop-down containing all the compatible poses. What if there is a lot of compatible poses? Would this affect the usability? Can this even happen in the real world?

**Comment by mwinokan** [comment]
@boriskovar-m2ms

I see. I think just add them all to the drop-down for now. I don't think there will be more than a dozen for any of the current data.

**Comment by mwinokan** [comment]
@mwinokan to think about how compound aliases (#1262) can elegantly be shown in the new redesigned UI

**Comment by mwinokan** [comment]
The aliases (#1262) are more important than the default Zinc ID's, so the hit navigator rows should show the compound alias and not Zinc ID. The Zinc ID should still be present in the download metadata if the user needs it, and perhaps it can be in the hover or clipboard if the alias is selected?

**Comment by mwinokan** [comment]
@matej-vavrek is back and is now working on this ticket

**Comment by phraenquex** [comment]
@mwinokan might be relevant to add something using the mol properties api #1177

**Comment by mwinokan** [comment]
@mwinokan to talk to Daren/curators to see how we can display useful alias and property information. The search function should also look for matches in this metadata

**Comment by mwinokan** [comment]
@matej-vavrek I had a look at your stack, really great progress!

Comparing the mockup:

<img width="730" alt="Screenshot 2024-05-29 at 13 10 27" src="https://github.com/m2ms/fragalysis-frontend/assets/36866506/05efcdda-e0a3-4470-aafb-33c139b84ff1">

to your draft:

<img width="730" alt="Screenshot 2024-05-29 at 13 10 27" src="https://github.com/m2ms/fragalysis-frontend/assets/36866506/6f0e78b6-5bbb-4986-b2a7-749996d80318">

I think the following improvements can be made:

- [ ] the observation shortcode (`AX????`) and compound string (`Z*`) should be on separate lines
- [ ] Please make the text size as big as possible (within the constraints of the UI), improving legibility will be a big win for the redesign
- [ ] If possible please take the colour of the Canonical and Conformer sites from the colour of their associated tags.
- [ ] The colour of the observations counter/button should be a colour scale from grey (1 observation) to bright orange (10+ observations)
- [ ] Please follow the above suggestions for the observations panel too
- [ ] The observations panel rows will also need the 'target'/'center on' buttons, not sure if this will need work from Boris.

I hope that helps!

**Comment by mwinokan** [comment]
@matej-vavrek please let me know which of the above tasks are technically difficult, we might want to move them to other tickets as the current version is very close to being ready for staging.

**Comment by matej-vavrek** [comment]
@mwinokan all should be doable. I am doing some final touches.

**Comment by phraenquex** [comment]
@mwinokan and @phraenquex need to update the tooltip spec

**Comment by mwinokan** [comment]
Summarising conversation w/ @matej-vavrek in slack:

- [ ] Hiding a tag that is currently selected in the tag details panel should **deselect** the tag
- [ ] The hit navigator and observation panel rows should still show **generated** tags even if they are hidden
- [ ] Hidden **curator** tags should not be shown in the hit navigator and observations panel

**Comment by mwinokan** [comment]
Thank you @matej-vavrek for implementing the above! My hopefully final nitpicks are:

- [ ] Some of the new components are not consistently aligned vertically (which looks a bit awkward):

<img width="463" alt="Screenshot 2024-06-11 at 16 41 29" src="https://github.com/m2ms/fragalysis-frontend/assets/36866506/24d389aa-0c99-4134-96e6-e95efde13808">

- [ ] Is it possible/easy to scale the text in each of the new generated tag boxes to use as much of the horizontal space as possible? I.e. can the`A` be a larger font size than the `F1H`? If this looks to messy it might be worth keeping the sizes consistent.

<img width="143" alt="Screenshot 2024-06-11 at 16 43 00" src="https://github.com/m2ms/fragalysis-frontend/assets/36866506/7c0339bc-4e9e-495a-be66-0b3e18b1e520">

- [ ] Please make the font style consistent between the pose name/compound and observation names

<img width="574" alt="Screenshot 2024-06-11 at 16 45 56" src="https://github.com/m2ms/fragalysis-frontend/assets/36866506/d91bd916-28c9-404a-83a3-f9c05856c6ee">

- [ ] Is there a reason why the observation/pose short codes are always in ALL CAPS? In the download all the files/metadata uses the format `Ax0152a` vs `AX0152A` that is shown in fragalysis. Can we change the f/e to match the `Ax0152a` please.

**Comment by mwinokan** [comment]
Looking at @matej-vavrek's stack, there are a couple 'nice to have' things

- [ ] The "Tag was automatically deselected..." warning appears even when a tag that was not selected is hidden
- [ ] Please change the hover tooltip for the observations counter/button to be `Show Observations`

**Comment by phraenquex** [comment]
Probably not this ticket, but there's still a bug in the XCA-upload-display flow, because these tags should not all be identical:

![image](https://github.com/m2ms/fragalysis-frontend/assets/33267937/7544f183-89d2-483b-919e-7dae244ab0df)

[This is the snapshot](https://fragalysis.xchem.diamond.ac.uk/viewer/react/projects/54/56) - though it's not redrawing properly - one for #1303, presumably already fixed.

**Comment by mwinokan** [comment]
Looking at the metadata.csv from CHIKV_Mac on staging:

* There is only one Crystalform in the dataset
* There is only one Quatassembly
* There are 14 different CrystalformSites
* There are 14 ConformerSites

Looking specifically at `cx1125[a-d]` there should be:

* 3 conformersites: 2a, 13a, 14a
* 3 Canonsites: 2, 13, 14

* 3 crystalformsites: F1a, F1b, F1c, 
* 1 Quatassembly: A1
* 1 Crystalform: F1

@phraenquex This seems to be correctly captured in the F/E. So there could be a bug earlier in the pipeline (upload) but it seems consistent to me. I will check with Jasmin. 

<img width="844" alt="Screenshot 2024-06-18 at 11 55 53" src="https://github.com/m2ms/fragalysis-frontend/assets/36866506/91c99f08-6fc7-44f4-874c-6c46b881aad1">

<img width="367" alt="Screenshot 2024-06-18 at 11 56 06" src="https://github.com/m2ms/fragalysis-frontend/assets/36866506/316f9506-3564-4dd2-bac2-cf276d764be5">

<img width="365" alt="Screenshot 2024-06-18 at 11 56 12" src="https://github.com/m2ms/fragalysis-frontend/assets/36866506/0b346633-b822-420e-9edc-71df90601d21">

**Comment by phraenquex** [comment]
It's the crystalform sites that don't show up correctly in the observations pane.

For cx1134a:  one observation (cx1134b) should be split into a new pose;  but cx1134c should (surely!?) be from a different crystalform site.  They can't both be from F1a.

![image](https://github.com/m2ms/fragalysis-frontend/assets/33267937/a657f3c3-70ba-48c8-8bf3-110947369961)

I'm guessing the loader is picking the wrong field from the yaml.

@ConorFWild can you please work with @kaliif to pin this down.

(@mwinokan please break into a separate ticket if appropriate.)

**Comment by phraenquex** [comment]
For this ticket:  we need some tricks (tooltips?) to make it quicker to work out why sites are assigned as they are.  

@mwinokan @ConorFWild please come bug me so we pin down the spec - it could be quite trivial.

**Comment by kaliif** [comment]
> It's the crystalform sites that don't show up correctly in the observations pane.
> 
> For cx1134a: one observation (cx1134b) should be split into a new pose; but cx1134c should (surely!?) be from a different crystalform site. They can't both be from F1a.

> I'm guessing the loader is picking the wrong field from the yaml.
> 
> @ConorFWild can you please work with @kaliif to pin this down.
> 
> (@mwinokan please break into a separate ticket if appropriate.)

They're not from the same crystalformsite, but those sites have the same crystalform and the tag is generated crystalform name, rather than crystalform site, hence F1a for all three.

But if [this](https://github.com/m2ms/fragalysis-frontend/issues/1277#issuecomment-1921258272) is the spec for the tags, then it seems that crystalformsite tag scheme is wrong in the backend, the loader seems to be naming crystalformsites using crystalform data. Have you noticed this? Are the tags just wrong?

**Comment by phraenquex** [comment]
Thanks @kaliif for digging - yes, that tag should definitely be generated with the `CrystalForm Site`.  So the bug is in the loader.

(We're talking about the middle (orange) one, right?)

I'm creating a this new ticket:  #1466

**Comment by phraenquex** [comment]
Reposting comment above, to confirm that IS in scope for this ticket: 
https://github.com/m2ms/fragalysis-frontend/issues/1322#issuecomment-2177955732

**Comment by mwinokan** [comment]
@phraenquex says there are outstanding issues with the tooltips:

Currently the tooltips show the tag category which is important:

<img width="539" alt="Screenshot 2024-06-20 at 11 29 09" src="https://github.com/m2ms/fragalysis-frontend/assets/36866506/f9e1e941-72f8-4426-b365-5198ef77ee8e">

But we also need to communicate the 'long site string' from XCA:

e.g.: `CrystalformSites - A71EV2A-x0450/A/201 - small cell` `<xtalform_sites.xtalform_site_id>`

@matej-vavrek please change the tooltips for the new XCA colourful buttons/boxes to include not only the tag type but also the longer site_id from XCA

**Comment by phraenquex** [comment]
Bugs: 

- [ ] If you change the main observation, the name of the pose doesn't update (in the LHS pane)
- [ ] "Select displayed hits" is broken - making it basically impossible to curate tags.  (I turned [cx0314b](https://fragalysis.xchem.diamond.ac.uk/viewer/react/projects/57/59) into a new pose, and it disappeared (or something) from the LHS. 
![image](https://github.com/m2ms/fragalysis-frontend/assets/33267937/bbdc4867-d891-4f1a-bc35-65b3400aaf8a)

@matej-vavrek please create new tickets as relevant, else move this back to "In progress."

(The hit navigation selection tools need a serious spec overhaul - @mwinokan a high priority for when you're back.)

**Comment by mwinokan** [comment]
@matej-vavrek please let us know which of the above are difficult. @phraenquex to help identify which ones are blockers for green

**Comment by matej-vavrek** [comment]
@phraenquex , @mwinokan , from the top of my head:

Need to investigate (and time):
- [ ] If you change the main observation, the name of the pose doesn't update (in the LHS pane)   @phraenquex to review spec.
- [ ] I turned [cx0314b](https://fragalysis.xchem.diamond.ac.uk/viewer/react/projects/57/59) into a new pose, but that pose never showed up from the LHS (as far as I can tell). ![image](https://github.com/m2ms/fragalysis-frontend/assets/33267937/bbdc4867-d891-4f1a-bc35-65b3400aaf8a)
- [ ] Only observation checkboxes to be used with the pose/name buttons.  (That would fix the weird name setting behaviour.) **[M: maybe auto-deselect selected observations on observation modal close could help/solve this]**
- [ ] Pose checkboxes should be detached from Observation checkboxes.  (The former are for feeding into computations, the latter for curating poses - so different use cases.  We may discover this is restrictive, but we'll worry about that later.) **[M: this is probably only a visual indication]**
- [ ] Funny jumping-around when setting the name.  (This may also be a checkbox bug.)

Should be alright:
- [ ] Compound id should be displayed for the observations, just like in the Hit pane for poses.
- [ ] Change "group" to "pose"
- [ ] Leave the LPC buttons visible, grey out if no selection boxes clicked.
- [ ] Open the new observations pane immediately when clicking on a different pose. (Currently closes it.)
- [ ] For poses:  make pose name bold, make compound name non-bold one point smaller, slightly greyer.
- [ ] For observations:  make the main observation bolder.  (And copy the compound name font etc from pose.)

More time consuming:
- [ ] Draw heavy blue border around entire pose for which the observation is open.  (Add spec to #1417 ) **[M: this could be also alright]**
- [ ] Auto-size the observations pane.  Stretch as big as needed for all observations.  Make it drag-resizable.  (Add spec to #1417) **[M: was this only for height?]**
- [ ] Need `select all` button in observations pane
- [ ] Need `select all displayed` button in obs pane.
- [ ] Multi-row tooltip, like this (ish): ![Image](https://github.com/m2ms/fragalysis-frontend/assets/33267937/76cd2ef9-ce4e-4dea-8831-d4d0ab540f24)

Backend related:
- [ ] Tooltips:  always singular (`site` not `sites`) **[M: we are getting tag categories names in plural from [backend API](https://fragalysis-matej-default.xchem-dev.diamond.ac.uk/api/tag_category/)]**

**Comment by matej-vavrek** [comment]
@phraenquex, @mwinokan, I have updated my stack with checked points from my comment. Could you have a look, please? There is also experimental "auto-deselect selected observations on observation modal close" but it is not and will not be a proper solution after discussion with Boris, please ignore this behaviour.

**Comment by mwinokan** [comment]
Thanks @matej-vavrek, most of it looks great. A couple notes:

- [ ] I am unable to select multiple observations, when I select a second one the first is deselected
- [ ] The lack of spacing between the observations modal and new dark-blue border is bothering me slightly, and if it's easy please remove the white padding between the existing hit navigator row border and the new dark-blue one:

![Screenshot 2024-07-11 at 09 09 25](https://github.com/m2ms/fragalysis-frontend/assets/36866506/867a471b-76fe-4394-a79c-3a0b8cfbff15)

Also, you were asking about being able to resize the observations modal, for #1476 we will need a wider modal anyway, I suggest increasing the size of the highlighted part below when the modal becomes wider:

![Screenshot 2024-07-11 at 09 11 32](https://github.com/m2ms/fragalysis-frontend/assets/36866506/75867dc0-4c1e-4dde-98f3-080a8344c6b5)

**Comment by matej-vavrek** [comment]
Thank you @mwinokan  for the feedback. I have updated my stack with fix. I am still not 100% sure if the border and spacing with modal is sufficient.

**Comment by mwinokan** [comment]
@matej-vavrek says auto-sizing only the height is easier, @phraenquex says that's fine. Width resizing can be revisited with combi-soaks #1476

**Comment by matej-vavrek** [comment]
I am unable update my stack today. Should the multi-row tooltip looks something like this? It shows on XCA tags hover.
![obrázok](https://github.com/user-attachments/assets/ff8df7ab-1035-44f8-81ca-37d2643216e1)

**Comment by phraenquex** [comment]
Thanks @matej-vavrek - as ever, a mockup helps so much!  Now I see how I mis-speced: it actually needs to be a table.

![image](https://github.com/user-attachments/assets/739f2df1-419c-4b14-ab96-817220b7b9b3)

So in each case, the bit after the dash is what needs to go into each cell.  (The short codes are already displayed in the coloured boxes.)

Auto-size the columns (and overall width?) so they never wrap (unless of course the tooltip ends up wider than the browser window).

**Comment by mwinokan** [comment]
@phraenquex can I suggest the table is in long format rather than wide? I think that would help reduce it's width:
|      Type       |          Value         |
|-----------------|------------------------|
| CanonSite       |  A71EV2A-x0211/A/120/A |
| ConformerSite   |  A71EV2A-x486          |
| CrystalformSite |  A71EV2A-x0486/A/147   |
| CrystalForm     |  A712A_open            |
| QuatAssembly    |  monomer               |

**Comment by matej-vavrek** [comment]
Current proposal (header row omitted after discussion with @mwinokan ):
![obrázok](https://github.com/user-attachments/assets/e18662e5-bb9e-405c-be24-a0142f9148dd)

**Comment by phraenquex** [comment]
No it needs to be wide:
* must must have the same orientation as the coloured boxes.
* for when there are many observations.  (The tooltip must show the strings for all observations in the pose.)

(It might be helpful to have a a vertically aligned single-observation tooltip when you hover somewhere (observation string?) - or it might be confusing.)

The CrystalFormSite needs to have the bit before th hyphen chopped off.

**Comment by mwinokan** [comment]
@matej-vavrek is working on the wide-format table

**Comment by matej-vavrek** [comment]
@phraenquex, @mwinokan, could it be something like this?
![obrázok](https://github.com/user-attachments/assets/8a126414-7ce2-4c7b-9149-315e2c8c919b)
![obrázok](https://github.com/user-attachments/assets/00aa28b0-c18a-4e13-a6d2-e4d3714e8374)

Or maybe try another colour for individual observations?
![obrázok](https://github.com/user-attachments/assets/ae2e0741-2107-4e7f-ba2b-6a3fcf1ab23f)

**Comment by mwinokan** [comment]
Thanks @matej-vavrek, I think the second one (without the overlap) looks good, but what if there is a large number of observations?

**Comment by matej-vavrek** [comment]
It is the same variant, but different observation 😅 If we do not want the overlap then we can maybe put it on top of the table?

**Comment by mwinokan** [comment]
@matej-vavrek ah I see. In that case could you maybe just highlight the row that is currently being hovered over? Either way I think the single wide table will be enough

**Comment by phraenquex** [comment]
@matej-vavrek - what does it look like with only the table?  I'm still pretty sure that'll be my preference - too much tooltip-ping gets confusing.

**Comment by matej-vavrek** [comment]
@phraenquex, only the table
![obrázok](https://github.com/user-attachments/assets/f3ac95c7-5fe0-4ede-86f2-90ae6f18b548)

With possible row highlighting - when you hover over an observation its row highlights
![obrázok](https://github.com/user-attachments/assets/79433230-b470-4cd6-b5f7-0fdde9e0060f)

**Comment by phraenquex** [comment]
@matej-vavrek let's go with only-table for now, then.  (We can review later.)

Leave out the highlight;  here's a better solution:  align the table rows horizontally with the observations in the panel.  (That way the eye can easily work out what is what.)


(Btw, this has immediately made it clear there's a bug in the loader, it's clearly pulling the wrong strings from `metadata.yaml`.  We'll have to discuss tomorrow with Tim if there's absolutely no way to get this fixed before Kalev is back.)

**Comment by phraenquex** [comment]
Yes, ideally:  as you scroll the observations, scroll the table.  (Is that easy to do?)

The table itself doesn't need a scroll-bar.

Also, if it's easy:  make the text in the tooltip selectable for copy-paste.

**Comment by phraenquex** [comment]
Yes you're right.  So put an active strip to the right, with a pop-out arrow (ilke on google maps).
![image](https://github.com/user-attachments/assets/1d637ad0-a733-4b06-9056-74d3dfe7d1ab)

**Comment by phraenquex** [comment]
Yes you're right.  But I think better than hover is to put an active strip to the right, with a pop-out arrow (ilke on google maps).

(If it's easy to do:  make it a pin, so that the pop-out happens (or doesn't) when you open the observation pane.)

**Comment by matej-vavrek** [comment]
> (If it's easy to do: make it a pin, so that the pop-out happens (or doesn't) when you open the observation pane.)

I am not sure if I understand this correctly @phraenquex  - instead of a pop-out arrow there will be a pop-out checkbox (or pin like icon?) and when this will be checked/set, then the Observations dialog will be shown in extended version and when it is not the it will be like it is now? So the state will be preserved and not lost on close/open?

**Comment by phraenquex** [comment]
@matej-vavrek yes correct.  Or better:  have both the arrow and the pin icon below one another: arrow just pops open the extension;  the pin pops it out and keeps it out.

But very important:  only do the pin if it's easy to do!!!  If not, put it into a separate ticket, and we can review later if it's required.

**Comment by matej-vavrek** [comment]
@phraenquex it should not be difficult but it will take some time to implement. I would prefer creating a new ticket so this one can be finished and merged.

Changes are on [my stack](https://fragalysis-matej-default.xchem-dev.diamond.ac.uk/), including copy button from #1465

**Comment by phraenquex** [comment]
@matej-vavrek when will the fix we discussed yesterday be visible/testable?

**Comment by matej-vavrek** [comment]
@phraenquex at the end of today or tomorrow morning

**Comment by matej-vavrek** [comment]
@phraenquex, I updated [my stack](https://fragalysis-matej-default.xchem-dev.diamond.ac.uk/). Is this going the right direction, please?
Currently, there is a problem with an column alignment and "header" in expanded window, because additional columns in each row are basically as separate parts..
![obrázok](https://github.com/user-attachments/assets/510070e8-dfcb-4107-9d91-f741336dd7f3)

**Comment by matej-vavrek** [comment]
@phraenquex, I have fixed position of the button and also overlaying layout. I can merge it if it is fine now.

**Comment by phraenquex** [comment]
@matej-vavrek that's perfect!!  Please merge it into staging ASAP - I need it to finalise spec for #1482 (green).

**Comment by matej-vavrek** [comment]
@phraenquex, it should be merged in staging now and I have also updated my stack with our latest stagingcandidate.

**Comment by mwinokan** [comment]
@phraenquex raises the issue that it is difficult to find the original crystallographic data from the f/e UI. 

@matej-vavrek please add the observation long code to make it easier to trace.

**Comment by matej-vavrek** [comment]
@phraenquex, @mwinokan, I have updated [my stack](https://fragalysis-matej-default.xchem-dev.diamond.ac.uk/) - added long code column and updated sorting for tags in "Tag Details".

Expanded observations dialog is getting quite wide now and I was thinking if in the long code column should not be just a copy button instead of the long string and the string put to the button's tooltip. What do you think?

Sorting in "Tag details" is same as in edit tags dialog so it sorts by category first and then by name. Reverse sorting reverts only name sorting and categories are sorted the same. Is this behaviour alright or should it do in other way..?

**Comment by mwinokan** [comment]
Thanks @matej-vavrek, the new long code field and sorting looks good to me. Please push to staging.

The text in the fields will get shorter (#1496) so I don't think you need to make changes.
